dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithEuTaxonomyNonFinancials.cy.ts

```diff
@@ -0,0 +1,181 @@
+      fixturesForTests,
+    ).then(() => {
+      const betaTotalAlignedCapexPercentage = roundNumber(
+        assertDefined(betaCapex.alignedShare?.relativeShareInPercent),
+        2,
+      );
+      const gammaCapexTotalAmountFormattedString = formatAmountWithCurrency({
+        amount: assertDefined(gammaCapexTotalAmount.value),
+        currency: assertDefined(gammaCapexTotalAmount.currency),
+      });
+      const gammaTotalAlignedCapexAbsoluteShareString = formatAmountWithCurrency(
+        assertDefined(gammaCapex.alignedShare?.absoluteShare),
+      );
+      const gammaContributionToClimateChangeMitigation = roundNumber(
+        assertDefined(gammaCapex.substantialContributionToClimateChangeMitigationInPercent),
+        2,
+      );
+
+      getSectionHead("Revenue").click();
+      getCellValueContainer("Total Amount", 0)
+        .invoke("text")
+        .should("match", /^0\s*EUR\s.*/);
+      getSectionHead("Revenue").click();
+
+      getSectionHead("CapEx").click();
+      getSectionHead("Aligned Share").click();
+      getCellValueContainer("Relative Share in Percent", 1)
+        .invoke("text")
+        .should("equal", `${betaTotalAlignedCapexPercentage} %`);
+      getCellValueContainer("Absolute Share", 2)
+        .invoke("text")
+        .should("equal", `${gammaTotalAlignedCapexAbsoluteShareString}`);
+      getCellValueContainer("Substantial Contribution to Climate Change Mitigation In Percent", 2)
+        .invoke("text")
+        .should("equal", `${gammaContributionToClimateChangeMitigation} %`);
+
+      getCellValueContainer("Total Amount", 2)
+        .invoke("text")
+        .should("match", new RegExp(`^${gammaCapexTotalAmountFormattedString}\\s.*$`));
+      getCellValueContainer("Total Amount", 2).click();
+      cy.contains("td", gammaCapexTotalAmountFormattedString).should("exist");
+      cy.contains("td", assertDefined(gammaCapexTotalAmount.quality).toString()).should("exist");
+      cy.contains("td", assertDefined(gammaCapexTotalAmount.comment).toString()).should("exist");
+      cy.get(`span[data-test="Report-Download-${assertDefined(gammaCapexTotalAmount.dataSource).fileName}"]`).should(
+        "exist",
+      );
+      cy.get("body").type("{esc}");
+
+      getCellValueContainer("Aligned Activities", 2).click();
+      cy.get("tr")
+        .contains("td", assertDefined(gammaCapexFirstAlignedActivity.activityName))
+        .nextAll()
+        .eq(4)
+        .invoke("text")
+        .should("match", /^0 %$/);
+      cy.get("body").type("{esc}");
+
+      getCellValueContainer("Non-Aligned Activities", 2).click();
+      cy.get("tr")
+        .contains("td", assertDefined(gammaCapexFirstNonAlignedActivity.activityName))
+        .nextAll()
+        .eq(2)
+        .invoke("text")
+        .should(
+          "match",
+          new RegExp(`^${assertDefined(gammaCapexFirstNonAlignedActivity.share).relativeShareInPercent}\\s.*$`),
+        );
+    });
+  });
```

Comment:
Meinst du das ist genug validation vom modal content? Das war ja vorher deutlich mehr.

Reply:
Soweit ich das gerade gesehen habe, wird in main das "non-aligned-acitvity" modal überhaupt nicht geöffnet und validiert. Ist also gar nicht gecovert aktuell (siehe `EuTaxonomyForNonFinancialsPanel.cy.ts` in main).

D.h. der Fakt dass ich es hier überhaupt mal aufmache und reinschaue ist schon mehr validation als in main.

Reply:
Ich hab mir den three-layer-datatable Test nochmal angeschaut.  Meine Einschätzung:

- An sich kann man schon Gründe dafür finden, dass man für eine Aligned-Activity bzw. Non-Aligned-Activity mal ALLE einzelnen Spalten in dem Modal auf die korrekte Anzeige ihres Wertes prüfen könnte.
Das würde halt zu 99% sicherstellen, dass einzelne Anzeigen nicht irgendwie kaputt sind.

- Genauso kann man aber Gründe dagegen finden:
1) z.B. dass man die Test-Philosophie vertritt, dass man nicht jede einzelne Möglichkeit abtesten will und das overkill hier wäre
2) dass das hier viele Zeilen Code erzeugen würde und auch eher das Modal im Detail testen würde als die Integration der MLDT für non-financials

Also würde ich mich v.a. zwecks Sprintende eher dagegen entscheiden.